client/src/components/ItemDetails/index.jsx

```diff
@@ -0,0 +1,382 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+// import icons
+import { IconContext } from 'react-icons';
+import {
+  MdKeyboardArrowUp,
+  MdKeyboardArrowDown,
+  MdAddShoppingCart,
+  MdPlaylistAdd,
+  MdDone,
+} from 'react-icons/md';
+import {
+  FaPinterest,
+  FaFacebookF,
+  FaTwitter,
+  FaCircleNotch,
+} from 'react-icons/fa';
+
+import InputField from '../common/InputField';
+
+import './ItemDetails.css';
+
+class ItemDetails extends Component {
+  state = {
+    cart: 'default',
+    wList: 'default',
+    color: '',
+    quantity: 0,
+    tags: [],
+  };
+
+  addToCartButton = {
+    adding: (
+      <Fragment>
+        <FaCircleNotch className="item-details__add-to-cart--icon-spin" />{' '}
+        ADDING ..
+      </Fragment>
+    ),
+    added: (
+      <Fragment>
+        <MdDone /> ITEM ADDED
+      </Fragment>
+    ),
+    default: (
+      <Fragment>
+        <MdAddShoppingCart /> ADD TO CART
+      </Fragment>
+    ),
+  };
+
+  addToWListButton = {
+    adding: <FaCircleNotch className="item-details__add-to-cart--icon-spin" />,
+    added: <MdDone />,
+    default: (
+      <IconContext.Provider value={{ size: '25px' }}>
+        <MdPlaylistAdd />
+      </IconContext.Provider>
+    ),
+  };
+
+  prevItem = () => {
+    // go to previous item
+  };
+
+  nextItem = () => {
+    // go to next item
+  };
+
+  handelChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
+  clearSelection = () => {
+    this.setState({ color: '' });
+  };
+
+  handelIncrement = e => {
+    const { quantity } = this.state;
+
+    if (e.target.className.includes('qty-inc')) {
+      this.setState(prevState => ({ quantity: prevState.quantity + 1 }));
+    } else if (e.target.className.includes('qty-dec') && quantity > 1) {
+      this.setState(prevState => ({ quantity: prevState.quantity - 1 }));
+    }
+  };
+
+  addToCart = () => {
+    const { details, userId, status } = this.props;
+    const { color, quantity } = this.state;
+    if (color && quantity && (!status || status === 'sale')) {
+      this.setState({ cart: 'adding' });
+      fetch('/api/v1/add-to-cart', {
```

Comment:
I prefer to use `'axios'`
```suggestion
     axios.post('/api/v1/add-to-cart', {
           userId,
          itemDetailId: details[Number(color)].id,
          quantity,
  })
  .then(...
```
, it's much easier, so no need for `credentials`, `JSON.stringify`, `headers`

Reply:
but I think we all used `fetch`